\hypertarget{classTestException}{}\doxysubsection{Test\+Exception Class Reference}
\label{classTestException}\index{TestException@{TestException}}


An exception thrown by a failed unit test.  




{\ttfamily \#include $<$Test\+Exception.\+h$>$}

\doxysubsubsection*{Public Member Functions}
\begin{DoxyCompactItemize}
\item 
\mbox{\hyperlink{classTestException_a890790e095bf4fc005765d7e48878ed2}{Test\+Exception}} ()
\begin{DoxyCompactList}\small\item\em Default constructor. \end{DoxyCompactList}\item 
\mbox{\hyperlink{classTestException_aea11191433977ed38c1e9a72a505bccc}{Test\+Exception}} (const char $\ast$function, const char $\ast$\mbox{\hyperlink{classTestException_a827fdb190ceb62cffd8979aa681e853c}{message}}, const char $\ast$file, int line)
\begin{DoxyCompactList}\small\item\em Constructor for throwing. \end{DoxyCompactList}\item 
\mbox{\hyperlink{classTestException_a4ab102f852629e2960120796708b0c5e}{Test\+Exception}} (const char $\ast$\mbox{\hyperlink{classTestException_a827fdb190ceb62cffd8979aa681e853c}{message}}, const char $\ast$file, int line)
\begin{DoxyCompactList}\small\item\em Constructor without function name parameter. \end{DoxyCompactList}\item 
\mbox{\hyperlink{classTestException_ac3b4f664f6c5370875f9bb21a46a96d8}{$\sim$\+Test\+Exception}} ()
\begin{DoxyCompactList}\small\item\em Destructor. \end{DoxyCompactList}\item 
void \mbox{\hyperlink{classTestException_a82eac00a19d3815d3e2f12c6ef99f824}{write}} (std\+::ostream \&out)
\begin{DoxyCompactList}\small\item\em Write error message to output stream. \end{DoxyCompactList}\item 
const std\+::string \& \mbox{\hyperlink{classTestException_a827fdb190ceb62cffd8979aa681e853c}{message}} ()
\begin{DoxyCompactList}\small\item\em Return the error message. \end{DoxyCompactList}\end{DoxyCompactItemize}
\doxysubsubsection*{Protected Attributes}
\begin{DoxyCompactItemize}
\item 
std\+::string \mbox{\hyperlink{classTestException_a94be8dc3502e2cf691bc39960239eadc}{message\+\_\+}}
\begin{DoxyCompactList}\small\item\em Error message. \end{DoxyCompactList}\end{DoxyCompactItemize}


\doxysubsubsection{Detailed Description}
An exception thrown by a failed unit test. 

Definition at line 19 of file Test\+Exception.\+h.



\doxysubsubsection{Constructor \& Destructor Documentation}
\mbox{\Hypertarget{classTestException_a890790e095bf4fc005765d7e48878ed2}\label{classTestException_a890790e095bf4fc005765d7e48878ed2}} 
\index{TestException@{TestException}!TestException@{TestException}}
\index{TestException@{TestException}!TestException@{TestException}}
\doxyparagraph{\texorpdfstring{TestException()}{TestException()}\hspace{0.1cm}{\footnotesize\ttfamily [1/3]}}
{\footnotesize\ttfamily Test\+Exception\+::\+Test\+Exception (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})}



Default constructor. 

Function Definitions -\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/-\/---. 

Definition at line 87 of file Test\+Exception.\+h.

\mbox{\Hypertarget{classTestException_aea11191433977ed38c1e9a72a505bccc}\label{classTestException_aea11191433977ed38c1e9a72a505bccc}} 
\index{TestException@{TestException}!TestException@{TestException}}
\index{TestException@{TestException}!TestException@{TestException}}
\doxyparagraph{\texorpdfstring{TestException()}{TestException()}\hspace{0.1cm}{\footnotesize\ttfamily [2/3]}}
{\footnotesize\ttfamily Test\+Exception\+::\+Test\+Exception (\begin{DoxyParamCaption}\item[{const char $\ast$}]{function,  }\item[{const char $\ast$}]{message,  }\item[{const char $\ast$}]{file,  }\item[{int}]{line }\end{DoxyParamCaption})}



Constructor for throwing. 

Constructs error message that includes file and line number. Values of the file and line parameters should be given by the built-\/in macros {\bfseries{F\+I\+LE}} and {\bfseries{L\+I\+NE}}, respectively, in the calling function. A typical call of the constructor is thus of the form\+: 
\begin{DoxyCode}{0}
\DoxyCodeLine{\textcolor{keywordflow}{throw} \mbox{\hyperlink{classTestException_a890790e095bf4fc005765d7e48878ed2}{TestException}}(\textcolor{stringliteral}{"MyClass::myFunction"}, \textcolor{stringliteral}{"A terrible thing happened!"},}
\DoxyCodeLine{                \_\_FILE\_\_, \_\_LINE\_\_ );}
\end{DoxyCode}



\begin{DoxyParams}{Parameters}
{\em function} & name of the function from which the \mbox{\hyperlink{classTestException}{Test\+Exception}} was thrown \\
\hline
{\em message} & message describing the nature of the error \\
\hline
{\em file} & name of the file from which the \mbox{\hyperlink{classTestException}{Test\+Exception}} was thrown \\
\hline
{\em line} & line number in file \\
\hline
\end{DoxyParams}


Definition at line 94 of file Test\+Exception.\+h.



References message(), and message\+\_\+.

\mbox{\Hypertarget{classTestException_a4ab102f852629e2960120796708b0c5e}\label{classTestException_a4ab102f852629e2960120796708b0c5e}} 
\index{TestException@{TestException}!TestException@{TestException}}
\index{TestException@{TestException}!TestException@{TestException}}
\doxyparagraph{\texorpdfstring{TestException()}{TestException()}\hspace{0.1cm}{\footnotesize\ttfamily [3/3]}}
{\footnotesize\ttfamily Test\+Exception\+::\+Test\+Exception (\begin{DoxyParamCaption}\item[{const char $\ast$}]{message,  }\item[{const char $\ast$}]{file,  }\item[{int}]{line }\end{DoxyParamCaption})}



Constructor without function name parameter. 


\begin{DoxyParams}{Parameters}
{\em message} & message describing the nature of the error \\
\hline
{\em file} & name of the file from which the \mbox{\hyperlink{classTestException}{Test\+Exception}} was thrown \\
\hline
{\em line} & line number in file \\
\hline
\end{DoxyParams}


Definition at line 116 of file Test\+Exception.\+h.



References message(), and message\+\_\+.

\mbox{\Hypertarget{classTestException_ac3b4f664f6c5370875f9bb21a46a96d8}\label{classTestException_ac3b4f664f6c5370875f9bb21a46a96d8}} 
\index{TestException@{TestException}!````~TestException@{$\sim$TestException}}
\index{````~TestException@{$\sim$TestException}!TestException@{TestException}}
\doxyparagraph{\texorpdfstring{$\sim$TestException()}{~TestException()}}
{\footnotesize\ttfamily Test\+Exception\+::$\sim$\+Test\+Exception (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})}



Destructor. 



Definition at line 134 of file Test\+Exception.\+h.



\doxysubsubsection{Member Function Documentation}
\mbox{\Hypertarget{classTestException_a82eac00a19d3815d3e2f12c6ef99f824}\label{classTestException_a82eac00a19d3815d3e2f12c6ef99f824}} 
\index{TestException@{TestException}!write@{write}}
\index{write@{write}!TestException@{TestException}}
\doxyparagraph{\texorpdfstring{write()}{write()}}
{\footnotesize\ttfamily void Test\+Exception\+::write (\begin{DoxyParamCaption}\item[{std\+::ostream \&}]{out }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [inline]}}



Write error message to output stream. 


\begin{DoxyParams}{Parameters}
{\em out} & output stream \\
\hline
\end{DoxyParams}


Definition at line 142 of file Test\+Exception.\+h.



References message\+\_\+.

\mbox{\Hypertarget{classTestException_a827fdb190ceb62cffd8979aa681e853c}\label{classTestException_a827fdb190ceb62cffd8979aa681e853c}} 
\index{TestException@{TestException}!message@{message}}
\index{message@{message}!TestException@{TestException}}
\doxyparagraph{\texorpdfstring{message()}{message()}}
{\footnotesize\ttfamily const std\+::string \& Test\+Exception\+::message (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [inline]}}



Return the error message. 



Definition at line 148 of file Test\+Exception.\+h.



References message\+\_\+.



Referenced by Unit\+Test\+Runner$<$ Unit\+Test\+Class $>$\+::method(), and Test\+Exception().



\doxysubsubsection{Member Data Documentation}
\mbox{\Hypertarget{classTestException_a94be8dc3502e2cf691bc39960239eadc}\label{classTestException_a94be8dc3502e2cf691bc39960239eadc}} 
\index{TestException@{TestException}!message\_@{message\_}}
\index{message\_@{message\_}!TestException@{TestException}}
\doxyparagraph{\texorpdfstring{message\_}{message\_}}
{\footnotesize\ttfamily std\+::string Test\+Exception\+::message\+\_\+\hspace{0.3cm}{\ttfamily [protected]}}



Error message. 



Definition at line 78 of file Test\+Exception.\+h.



Referenced by message(), Test\+Exception(), and write().



The documentation for this class was generated from the following file\+:\begin{DoxyCompactItemize}
\item 
Test\+Exception.\+h\end{DoxyCompactItemize}
